Homework/src/io/codelex/arithmetic/practice/CalculateArea.java

```diff
@@ -0,0 +1,153 @@
+        System.out.println("1. Calculate the Area of a Circle");
+        System.out.println("2. Calculate the Area of a Rectangle");
+        System.out.println("3. Calculate the Area of a Triangle");
+        System.out.println("4. Quit\n");
+        System.out.print("Enter your choice (1-4) : ");
+
+        // get input from user
+        userChoice = keyboard.nextInt();
+
+        // validate input
+        while (userChoice < 1 || userChoice > 4) {
+            System.out.print("Please enter a valid range: 1, 2, 3, or 4: ");
+            userChoice = keyboard.nextInt();
+        }
+
+        return userChoice;
+    }
+
+    public static void calculateCircleArea() {
+
+        BigDecimal radius = ZERO;
+
+        // Get input from user
+        Scanner keyboard = new Scanner(System.in);
+
+        do{
+            System.out.print("What is the circle's radius? ");
+            try{
+                radius=keyboard.nextBigDecimal();
+                break;
+            }catch(Exception e){
+                System.out.println("Bad input! Try again.");
+                keyboard.next();
+            }
+        }while(true);
+
+        // Display output
+        System.out.println("The circle's area is "
+                + Geometry.areaOfCircle(radius));
+    }
+
+    public static void calculateRectangleArea() {
+        BigDecimal length = ZERO;
+        BigDecimal width = ZERO;
+
+        // Get input from user
+        Scanner keyboard = new Scanner(System.in);
+
+        // Get length
+
+        do{
+            System.out.print("Enter length? ");
+            try{
+                length=keyboard.nextBigDecimal();
+                break;
+            }catch(Exception e){
+                System.out.println("Bad input! Try again.");
+                keyboard.next();
+            }
+        }while(true);
+
+        // Get width
+        do{
+            System.out.print("Enter width? ");
+            try{
+                width=keyboard.nextBigDecimal();
+                break;
+            }catch(Exception e){
+                System.out.println("Bad input! Try again.");
+                keyboard.next();
+            }
+        }while(true);
+
+        // Display output
+        System.out.println("The rectangle's area is "
+                + Geometry.areaOfRectangle(length, width));
+    }
+
+    public static void calculateTriangleArea() {
+        BigDecimal base = ZERO;
+        BigDecimal height = ZERO;
+
+        // Get input from user
+        Scanner keyboard = new Scanner(System.in);
+
+        // Get the base
+
+        do{
+            System.out.print("Enter length of the triangle's base? ");
+            try{
+                base=keyboard.nextBigDecimal();
+                break;
+            }catch(Exception e){
+                System.out.println("Bad input! Try again.");
+                keyboard.next();
+            }
+        }while(true);
+
+        // Get the height
+
+        do{
```

Comment:
Šis vairāk kā ieteikums, nav obligāti jālabo, jo tad vēl par metodēm nerunājām:
Šāds while cikls atkārtojas vairākas reizes. Daudz ērtāk būtu uztaisīt vienu metodi, kura sauktos, piemēram, getBigDecimalFromInput, kurā tad būtu tas cikls un tikko lietotājs ievada korektu vērtību - metode atgriež to skaitli. To metodi tad varētu izmantot visās vietās.